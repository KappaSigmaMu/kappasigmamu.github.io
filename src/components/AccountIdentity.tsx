import type { ApiPromise } from '@polkadot/api'
import type { DeriveAccountRegistration } from '@polkadot/api-derive/types'
import { AccountId } from '@polkadot/types/interfaces'
import { useEffect, useState } from 'react'
import { truncateMiddle } from '../helpers/truncate'

const AccountIdentity = ({
  accountId,
  api,
  hideNotSet
}: {
  accountId: AccountId
  api: ApiPromise
  hideNotSet?: boolean
}) => {
  const [id, setId] = useState<string>('')

  useEffect(() => {
    api.derive.accounts.identity(accountId, (identity: DeriveAccountRegistration) => {
      identity.display && setId(identity.display)
    })
  }, [])

  if (!id && hideNotSet) return <></>
  if (id) return <>{id}</>
  return <>{truncateMiddle(accountId?.toString(), 20)}</>
}

export { AccountIdentity }
