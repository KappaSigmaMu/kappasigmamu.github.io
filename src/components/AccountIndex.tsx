import type { AccountIndex as PAccountIndex } from '@polkadot/types/interfaces'
import { useEffect, useState } from 'react'
import { useKusama } from '../kusama'

const AccountIndex = ({ member } : { member: SocietyMember }) => {
  const { api } = useKusama()
  const [index, setIndex] = useState<string>('')

  useEffect(() => {
    api?.derive.accounts.idToIndex(member.accountId, (accountIndex: PAccountIndex) => {
      if (accountIndex) {
        const index = api.registry.createType('AccountIndex', accountIndex.toNumber()).toString()
        setIndex(index)
      }
    })
  }, [])

  return <>{index}</>
}

export { AccountIndex }
