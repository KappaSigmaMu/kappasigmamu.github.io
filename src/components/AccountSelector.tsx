import Identicon from '@polkadot/react-identicon'
import { Dropdown, DropdownButton, Spinner } from 'react-bootstrap'
import styled from 'styled-components'
import { useAccount } from '../account/AccountContext'
import { truncateMiddle } from '../helpers/truncate'
import { useKusama } from '../kusama'

interface LevelStatusType {
  [key: string]: string
}

const LEVELSTATUS: LevelStatusType = {
  human: 'WAITING BID',
  bidder: 'BID SUBMITTED',
  candidate: 'WAITING POI',
  cyborg: ''
}

const Title = ({ activeAccount }: { activeAccount: accountType }) => {
  const { level } = useAccount()
  const account = activeAccount.name

  // TODO: Title not working
  return (
    <>
      <div className="align-top d-inline-block me-3" style={{ marginTop: 6 }}>
        <Identicon value={activeAccount.address} size={32} theme={'polkadot'} />
      </div>

      <label style={{ fontSize: '12px' }}>
        <SelectedAccountDiv className="text-start mb-1">{truncateMiddle(account || '')}</SelectedAccountDiv>
        <LevelStatusDiv>
          <label className="pe-3">JOURNEY: {level.toUpperCase()}</label>
          <label>{LEVELSTATUS[level]}</label>
        </LevelStatusDiv>
      </label>
    </>
  )
}

const Main = () => {
  const { activeAccount, setActiveAccount, accounts } = useAccount()

  const onChange = (account: string) => {
    const activeAccount = accounts.filter((acc) => acc.address.includes(account))[0]
    setActiveAccount(activeAccount)
  }

  return (
    <StyledDropdownButton
      variant="outline-light-grey"
      onSelect={(eventKey: string) => onChange(eventKey)}
      title={<Title activeAccount={activeAccount} />}
    >
      {accounts.map((option: accountType) => (
        <Dropdown.Item style={{ fontSize: '12px' }} eventKey={option.address} key={option.address} href="#">
          {option.name ? option.name : option.address}
        </Dropdown.Item>
      ))}
    </StyledDropdownButton>
  )
}

const AccountSelector = () => {
  const { api } = useKusama()
  const loading = !api?.query
  return loading ? <Spinner animation="border" variant="primary" /> : <Main />
}

const LevelStatusDiv = styled.div`
  display: flex;
  justify-content: space-between;

  label {
    color: ${(props) => props.theme.colors.grey};
    font-weight: 700;
  }
`

const SelectedAccountDiv = styled.div`
  color: ${(props) => props.theme.colors.primary};
  font-weight: 700;
  font-size: 14px;
`

const StyledDropdownButton = styled(DropdownButton)`
  background-color: ${(props) => props.theme.colors.darkGrey};
  & ::after {
    margin-bottom: 10px;
  }
`

export { AccountSelector }
