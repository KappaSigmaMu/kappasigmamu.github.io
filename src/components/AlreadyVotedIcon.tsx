import { FaCheckDouble } from 'react-icons/fa6'
import { styled } from 'styled-components'

const AlreadyVotedIcon = () => (
  <>
    <StyledCheckDouble className="me-2" />
    <label style={{ color: '#6c757d' }}>Voted</label>
  </>
)

const StyledCheckDouble = styled(FaCheckDouble)`
  flex-shrink: 0;

  & path {
    fill: ${(props) => props.theme.colors.lightGrey};
  }
`

export { AlreadyVotedIcon }
