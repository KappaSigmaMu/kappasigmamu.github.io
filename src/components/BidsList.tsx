import Identicon from '@polkadot/react-identicon'
import { Col, Row } from 'react-bootstrap'
import styled from 'styled-components'
import { humanizeBidKind, humanizeBidValue } from '../helpers/humanize'
import { truncateMiddle } from '../helpers/truncate'

const Header = () => (
  <StyledHeaderRow>
    <Col xs={1} className="text-center">#</Col>
    <Col xs={4} className="text-start">Wallet Hash</Col>
    <Col xs={5} className="text-start">Bid Kind</Col>
    <Col xs={2} className="text-end">Value</Col>
  </StyledHeaderRow>
)

const DataRow = ({ bid }: { bid: any }) => (
  <StyledDataRow key={bid.who?.toString()}>
    <Col xs={1} className="text-center">
      <Identicon value={bid.who} size={32} theme={'polkadot'} />
    </Col>
    <Col xs={4} className="text-start text-truncate">
      {truncateMiddle(bid.who?.toString())}
    </Col>
    <Col xs={5} className="text-start text-truncate">
      {humanizeBidKind(bid.kind)}
    </Col>
    <Col xs={2} className="text-end">
      {humanizeBidValue(bid.kind)}
    </Col>
  </StyledDataRow>
)

const BidsList = ({ bids }: { bids: any }): JSX.Element => (
  <>
    <Header />
    {bids.map((bid: any) => (
      <DataRow key={bid.who} bid={bid} />
    ))}
  </>
)

const StyledHeaderRow = styled(Row)`
  color: #fff;
  line-height: 3;

  & .text-end {
    padding-right: 36px;
  }
`

const StyledDataRow = styled(StyledHeaderRow)`
  background-color: #343A40;
  border-radius: 6px;
  margin-top: 10px;
`

export { BidsList }
