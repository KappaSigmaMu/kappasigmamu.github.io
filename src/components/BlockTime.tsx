import type { BlockNumber } from '@polkadot/types/interfaces'
import { formatNumber } from '@polkadot/util'
import { useBlockTime } from '../hooks/useBlockTime'
import { useKusama } from '../kusama'

const BlockTime = ({ block }: { block: BlockNumber }) => {
  const { api } = useKusama()
  const [, time] = useBlockTime(block, api)

  return (<>
    <span>{time}</span>
    &nbsp;
    <span>(#{formatNumber(block)})</span>
  </>)
}

export { BlockTime }
