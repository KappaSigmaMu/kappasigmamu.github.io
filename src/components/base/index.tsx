import { Button, Row } from 'react-bootstrap'
import styled from 'styled-components'

const DataHeaderRow = styled(Row)`
  line-height: 3;

  & .text-end {
    padding-right: 36px;
  }

  * {
    color: ${(props) => props.theme.colors.lightGrey};
  }
`

const DataRow = styled(DataHeaderRow)`
  background-color: ${(props) => props.theme.colors.darkGrey};
  border-radius: 6px;
  margin-top: 10px;

  * {
  color: ${(props) => props.theme.colors.white};
}
`

const ExternalLink = styled.a.attrs(() => ({
  target: '_blank',
  rel: 'noreferrer'
}))``

const PrimaryButton = styled(Button).attrs(() => ({ variant: 'primary' }))``
const PrimaryLgButton = styled(PrimaryButton).attrs(() => ({ size: 'lg' }))``

const SecondaryLgButton = styled(Button).attrs(() => ({ variant: 'secondary', size: 'lg' }))``

export {
  DataHeaderRow,
  DataRow,
  ExternalLink,
  PrimaryButton,
  PrimaryLgButton,
  SecondaryLgButton
}
