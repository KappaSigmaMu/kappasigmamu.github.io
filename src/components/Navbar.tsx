import { Button } from 'react-bootstrap'
import Container from 'react-bootstrap/Container'
import Nav from 'react-bootstrap/Nav'
import { default as BNavbar } from 'react-bootstrap/Navbar'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { useAccount } from '../account/AccountContext'
import { fetchAccounts } from '../helpers/fetchAccounts'
import KappaSigmaMu from '../static/kappa-sigma-mu-logo.svg'
import { AccountSelector } from './AccountSelector'
import { SocialIcons } from './SocialIcons'

const Navbar = ({
  showAccount=false,
  showBrandIcon=false,
  showGalleryButton=false,
  showSocialIcons=false,
}: NavRouteProps) => (
  <BNavbar>
    <NavbarContainer>
      <Nav>{showBrandIcon ? <NavbarBrand /> : <></>}</Nav>
      <CenterNav>
        {showGalleryButton ? <Nav.Link to="/home/bids" as={Link}>Gallery</Nav.Link> : <></>}
        {showSocialIcons ? <SocialIcons /> : <></>}
        {showAccount ? <AccountNavbar /> : <></>}
      </CenterNav>
    </NavbarContainer>
  </BNavbar>
)

const NavbarBrand = () => (
  <BNavbar.Brand as={Link} to="/">
    <img width={90} src={KappaSigmaMu} alt="KappaSigmaMu Logo" />
  </BNavbar.Brand>
)

const AccountNavbar = () => {
  const { activeAccount, setActiveAccount, accounts, setAccounts } = useAccount()

  return accounts.length != 0 && activeAccount
    ? <AccountSelector />
    : (
      <Button variant="outline-secondary" onClick={() => fetchAccounts(setAccounts, setActiveAccount)}>
        Connect Wallet
      </Button>
    )
}

const NavbarContainer = styled(Container)`
  align-items: end !important;
  padding-top: 20px;
`

const CenterNav = styled(Nav)`
  align-items: center;
  align-self: normal;
`

export { Navbar }
