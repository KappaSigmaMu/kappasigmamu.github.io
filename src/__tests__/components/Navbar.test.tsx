import { BrowserRouter } from 'react-router-dom'
import TestRenderer from 'react-test-renderer'
import { Navbar } from '../../components/Navbar'

const defaultProps = {
  accounts: [],
  activeAccount: '5DcN2feEKzC23toLBu63N7Q9E2Tc3HE44oyd992WY31iZee4',
  setAccounts: () => ({}),
  setActiveAccount: () => ({}),
}

describe('NavBar', () => {
  it('renders it with showAccount', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Navbar {...defaultProps} showAccount />
      </BrowserRouter>
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders it with showBrandIcon', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Navbar {...defaultProps} showBrandIcon />
      </BrowserRouter>
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders it with showGalleryButton', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Navbar {...defaultProps} showGalleryButton />
      </BrowserRouter>
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders it with showSocialIcons', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Navbar {...defaultProps} showSocialIcons />
      </BrowserRouter>
    )

    expect(tree).toMatchSnapshot()
  })
})
