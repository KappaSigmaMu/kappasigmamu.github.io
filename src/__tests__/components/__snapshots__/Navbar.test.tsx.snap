// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders it with showAccount 1`] = `
<nav
  className="navbar navbar-expand navbar-light"
>
  <div
    className="sc-jSFjdj jmcpqm container"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    />
    <div
      className="sc-gKAaRy kWJcFL navbar-nav"
      onKeyDown={[Function]}
    >
      <button
        className="btn btn-outline-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Connect Wallet
      </button>
    </div>
  </div>
</nav>
`;

exports[`NavBar renders it with showBrandIcon 1`] = `
<nav
  className="navbar navbar-expand navbar-light"
>
  <div
    className="sc-jSFjdj jmcpqm container"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="KappaSigmaMu Logo"
          src="kappa-sigma-mu-logo.svg"
          width={90}
        />
      </a>
    </div>
    <div
      className="sc-gKAaRy kWJcFL navbar-nav"
      onKeyDown={[Function]}
    />
  </div>
</nav>
`;

exports[`NavBar renders it with showGalleryButton 1`] = `
<nav
  className="navbar navbar-expand navbar-light"
>
  <div
    className="sc-jSFjdj jmcpqm container"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    />
    <div
      className="sc-gKAaRy kWJcFL navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rr-ui-event-key={null}
        disabled={false}
        href="/home/bids"
        onClick={[Function]}
      >
        Gallery
      </a>
    </div>
  </div>
</nav>
`;

exports[`NavBar renders it with showSocialIcons 1`] = `
<nav
  className="navbar navbar-expand navbar-light"
>
  <div
    className="sc-jSFjdj jmcpqm container"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    />
    <div
      className="sc-gKAaRy kWJcFL navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="sc-dlnjwi"
        href="https://discord.gg/9AWjTf8wSk"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Discord Logo"
          src="discord-logo.svg"
        />
      </a>
       
      <a
        className="sc-dlnjwi"
        href="https://matrix.to/#/!BUmiAAnAYSRGarqwOt:matrix.parity.io?via=matrix.parity.io"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Element Logo"
          src="element-logo.svg"
        />
      </a>
       
      <a
        className="sc-dlnjwi"
        href="https://twitter.com/network"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Twitter Logo"
          src="twitter-logo.svg"
          style={
            Object {
              "color": "red",
            }
          }
        />
      </a>
    </div>
  </div>
</nav>
`;
